src/axom/primal/operators/winding_number.hpp

```diff
@@ -0,0 +1,391 @@
+{
+  bool isOnFace;
+  return winding_number(q, tri, isOnFace, edge_tol, EPS);
+}
+
+/*!
+ * \brief Computes the solid angle winding number for a 3D planar polygon
+ *
+ * \param [in] query The query point to test
+ * \param [in] poly The Polygon object
+ * \param [in] isOnFace Return variable to show if the point is on the polygon
+ * \param [in] edge_tol The physical distance level at which objects are 
+ *                      considered indistinguishable
+ * \param [in] EPS Miscellaneous numerical tolerance level for nonphysical distances
+ * 
+ * \pre Assumes the polygon is truly planar
+ * Triangulates the polygon and computes the triangular solid angle for each part
+ * 
+ * \return float the generalized winding number.
+ */
+template <typename T>
+double winding_number(const Point<T, 3>& q,
+                      const Polygon<T, 3>& poly,
+                      bool& isOnFace,
+                      const double edge_tol = 1e-8,
+                      const double EPS = 1e-8)
+{
+  const int num_verts = poly.numVertices();
+  if(num_verts < 3) return 0;
+
+  double wn = 0.0;
+  for(int i = 0; i < num_verts - 2; ++i)
+    wn += winding_number(q,
+                         Triangle<T, 3>(poly[0], poly[i + 1], poly[i + 2]),
+                         isOnFace,
+                         edge_tol,
+                         EPS);
+
+  return wn;
+}
+
+/*!
+ * \brief Computes the solid angle winding number for a 3D planar polygon
+ *
+ * Overload function without additional returning parameter
+ */
+template <typename T>
+double winding_number(const Point<T, 3>& q,
+                      const Polygon<T, 3>& poly,
+                      const double edge_tol = 1e-8,
+                      const double EPS = 1e-8)
+{
+  bool isOnFace;
+  return winding_number(q, poly, isOnFace, edge_tol, EPS);
+}
+
+/*!
+ * \brief Computes the solid angle winding number for a 3D convex polyhedron
+ *
+ * \param [in] query The query point to test
+ * \param [in] poly The Polyhedron object
+ * \param [in] useStrictInclusion If true, points on the boundary are considered exterior.
+ * \param [in] edge_tol The physical distance level at which objects are 
+ *                      considered indistinguishable
+ * \param [in] EPS Miscellaneous numerical tolerance level for nonphysical distances
+ * 
+ * \pre Expects the polyhedron to be convex
+ * 
+ * Computes the faces of the polygon and computes the winding number for each.
+ *
+ * \return int The integer winding number.
+ */
+template <typename T>
+int winding_number(const Point<T, 3>& query,
+                   const Polyhedron<T, 3>& poly,
+                   bool useStrictInclusion = false,
+                   double edge_tol = 1e-8,
+                   double EPS = 1e-8)
+{
+  SLIC_ASSERT(poly.hasNeighbors());
+  const int num_verts = poly.numVertices();
+
+  int* faces = new int[num_verts * num_verts];
+  int* face_size = new int[2 * num_verts];
+  int* face_offset = new int[2 * num_verts];
```

Comment:
This is only the way that it is so that the data is compatible with the later call to `Polyhedron::getFaces`. Is there a way to access an underlying integer array from an `axom::Array<int>`?

I'm also not opposed to working towards changing/overloading `Polyhedron::getFaces` to take in `axom::Array` arguments, if that's something that would be useful.

Reply:
I think `Array::data()` should do the trick

https://github.com/LLNL/axom/blob/31ae2e4787a602d9a7b44eea76c3130a30c8ce3d/src/axom/core/Array.hpp#L339-L340

Reply:
Ah, I was searching in `Array.hpp` for something with `get` in the name. I'll make the change, thank you!